manager/src/main/java/org/openremote/manager/rules/JsonRulesBuilder.java

```diff
@@ -800,8 +804,21 @@ protected RuleActionExecution buildRuleActionExecution(JsonRule rule, RuleAction
                 if (isPush) {
                     PushNotificationMessage pushNotificationMessage = (PushNotificationMessage) notification.getMessage();
                     body = pushNotificationMessage.getBody();
+                    action = pushNotificationMessage.getAction();
+
+                    if (action != null && action.getUrl() != null) {
+                        String newUrl = replaceAssetIdPlaceholder(action.getUrl(), ruleState, useUnmatched, "notification URL", true);
+                        action.setUrl(newUrl);
+                    }
```

Comment:
Add `pushNotificationMessage.setAction()` to apply these changes. 😉 
Similar as you did with the body below.